presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -240,8 +240,10 @@ public PlanNode visitGroupId(GroupIdNode node, RewriteContext<Expression> contex
         @Override
         public PlanNode visitMarkDistinct(MarkDistinctNode node, RewriteContext<Expression> context)
         {
-            checkState(!SymbolsExtractor.extractUnique(context.get()).contains(node.getMarkerSymbol()), "predicate depends on marker symbol");
-            return context.defaultRewrite(node, context.get());
+            if (!SymbolsExtractor.extractUnique(context.get()).contains(node.getMarkerSymbol())) {
```

Comment:
> distinct marker used in predicate

I understand this broader set of working queries, perhaps new test would be good.

Reply:
Not really. Typically `MarkDistinct` is used to provide distinct source values (let say column `A`)  to aggregate function. In order push down predicate on column `A` over `MarkDistinct` you would need to push down it over aggregation first, so that would mean that column `A` would must to survive the aggregation which is not possible. 
See: `SELECT count(DISTINCT name), count(DISTINCT regionkey) FROM region HAVING regionkey < 3;`

Filtered aggregations are not yet supported for aggregation with `DISTINCT`. Though I would expect them to filter column before marking distinct values.